pkg/upgrade/control_plane.go

```diff
@@ -458,6 +458,9 @@ func (u *ControlPlaneUpgrader) updateMachine(replacementKey ctrlclient.ObjectKey
 		desiredVersion := u.desiredVersion.String()
 		replacementMachine.Spec.Version = &desiredVersion
 
+		// clear status to ensure the provider can't match the new machine to any existing machine
```

Comment:
@andrewsykim I don't believe this is necessary. The apiserver will throw away anything in the .status portion of the object as part of the create flow. Are you seeing otherwise?

Reply:
I am seeing otherwise but it may be CAPV doing that and not the apiserver -- will verify in a bit

Reply:
yeah so turns out it was CAPV re patching the status there so we don't need this, thanks!

/close